spatial/r2/triangle_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r2
+
+import (
+	"math"
+	"testing"
+
+	"golang.org/x/exp/rand"
+)
+
+func TestTriangleDegenerate(t *testing.T) {
```

Comment:
I think this can be done more clearly by constructing triangles with known degeneracy from vector projection from a random line.

Reply:
Define a pair of points, then pick a middle and project a vector of known length perpendicular to the line at the middle point for the third vertex.

Reply:
Aha, that way we test for the "cap" type of degenerate triangle. Alright, I'll add tests for "cap" and "needle" degenerate triangles.